clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -231,26 +231,18 @@ data Declaration
   -- * List of parameters for this component (param name, param type, param value)
   --
   -- * Ports (port name, port direction, type, assignment)
-  | BlackBoxD
-      -- Primitive name:
-      !Text
-      -- VHDL only: add /library/ declarations:
-      [BlackBoxTemplate]
-      -- VHDL only: add /use/ declarations:
-      [BlackBoxTemplate]
-      -- Intel/Quartus only: create a /.qsys/ file from given template:
-      [((Text,Text),BlackBox)]
-      -- Template tokens:
-      !BlackBox
-      -- Context in which tokens should be rendered:
-      BlackBoxContext
-  -- ^ Instantiation of blackbox declaration
-  | NetDecl'
-      (Maybe Comment)            -- Note; will be inserted as a comment in target hdl
-      WireOrReg                  -- Wire or register
-      !Identifier                -- Name of signal
-      (Either Identifier HWType) -- Pointer to type of signal or type of signal
-      -- ^ Signal declaration
+  | BlackBoxD  -- ^ Instantiation of blackbox declaration
+      !Text                    -- ^ Primitive name
+      [BlackBoxTemplate]       -- ^ VHDL only: add @library@ declarations
+      [BlackBoxTemplate]       -- ^ VHDL only: add @use@ declarations
+      [((Text,Text),BlackBox)] -- ^ Intel Quartus only: create a @.qsys@ file from given template
+      !BlackBox                -- ^ Template tokens
+      BlackBoxContext          -- ^ Context in which tokens should be rendered
+  | NetDecl'  -- ^ Signal declaration
+      (Maybe Comment)            -- ^ Note; will be inserted as a comment in target hdl
+      WireOrReg                  -- ^ Wire or register
+      !Identifier                -- ^ Name of signal
+      (Either Identifier HWType) -- ^ Pointer to type of signal or type of signal
```

Comment:
Don't add `^` for fields. Haddock will trip over these if I'm not mistaken..

Reply:
My haddock uses them, and generates nicely documented fields

Reply:
Oh really, that's nice. I'm quite sure that didn't used to be the case. Nevermind my comment then.

Reply:
You're right :frowning_face:
This doesn't work the haddock <=2.20 (that come with GHC <= 8.4)

But it does work with haddock-2.22.0, (comes with GHC-8.6)

Reply:
Add CPP flags!

(/jk)